File: Python/977_Squares_of_a_Sorted_Array.py
"""
Given an integer array nums sorted in non-decreasing order, return an array of the squares of each number sorted in non-decreasing order.
"""
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        res = [None] * len(nums)
        left, right, pos = 0, len(nums) - 1, len(nums) - 1
        while left <= right:
            if abs(nums[left]) > abs(nums[right]):
                res[pos] = nums[left] * nums[left]
                left += 1
            else:
                res[pos] = nums[right] * nums[right]
                right -= 1   
            pos -= 1    
        return res               

File: Java/977_Squares_of_a_Sorted_Array.java
/**
 * Given an integer array nums sorted in non-decreasing order, return an array of the squares of each number sorted in non-decreasing order.
 * 
 */

 class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] res = new int[nums.length];
        int left = 0, right = nums.length - 1, len = nums.length - 1;
        while (left <= right) {
            if (Math.abs(nums[left]) > Math.abs(nums[right])){
                res[len] = nums[left] * nums[left];
                left++;
            } else {
                res[len] = nums[right] * nums[right];
                right--;
            }
            --len;
        }
        return res;
    }
}